src/main/kotlin/org/elm/lang/core/types/Ty.kt

```diff
@@ -72,9 +75,23 @@ data class TyRecord(
     override fun toString(): String {
         val f = fields.toString().let { it.substring(1, it.lastIndex) }
         return alias?.let {
-            "{${it.name}${if (it.parameters.isEmpty()) "" else " ${it.parameters.joinToString(" ")}"}}"
+            val prefix = if (fieldReferences.isEmpty()) "" else "+"
+            val params = if (it.parameters.isEmpty()) "" else " ${it.parameters.joinToString(" ")}"
+            "{$prefix${it.name}$params}"
         } ?: baseTy?.let { "{$baseTy | $f}" } ?: "{$f}"
     }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is TyRecord) return false
+        return fields == other.fields && baseTy == other.baseTy
```

Comment:
I'm sure you know what you're doing, but just to double-check, should `alias` be included in equality/hash?

Reply:
It doesn't actually affect anything, since only TyVars are used as map keys, and I never compare tys by equality. But it's probably better to add it for consistency with other data class tys.